src/VirtoCommerce.PricingModule.Data/Services/PricingEvaluatorService.cs

```diff
@@ -0,0 +1,180 @@
+        private readonly IPricingPriorityFilterPolicy _pricingPriorityFilterPolicy;
+        private readonly IItemService _productService;
+
+        public PricingEvaluatorService(
+                Func<IPricingRepository> repositoryFactory,
+                IItemService productService,
+                ILogger<PricingEvaluatorService> logger,
+                IPlatformMemoryCache platformMemoryCache,
+                IPricingPriorityFilterPolicy pricingPriorityFilterPolicy
+            )
+        {
+            _platformMemoryCache = platformMemoryCache;
+            _repositoryFactory = repositoryFactory;
+            _logger = logger;
+            _pricingPriorityFilterPolicy = pricingPriorityFilterPolicy;
+            _productService = productService;
+        }
+
+
+        public virtual async Task<IEnumerable<Pricelist>> EvaluatePriceListsAsync(PriceEvaluationContext evalContext)
+        {
+            var cacheKey = CacheKey.With(GetType(), nameof(EvaluatePriceListsAsync));
+            var priceListAssignments = await _platformMemoryCache.GetOrCreateExclusiveAsync(cacheKey, async cacheEntry =>
+            {
+                cacheEntry.AddExpirationToken(GenericCachingRegion<PricelistAssignment>.CreateChangeToken());
+
+                return await GetAllPricelistAssignments();
+            });
+
+            var query = priceListAssignments.AsQueryable();
+
+            if (evalContext.CatalogId != null)
+            {
+                query = query.Where(x => x.CatalogId == evalContext.CatalogId);
+            }
+
+            if (evalContext.Currency != null)
+            {
+                query = query.Where(x => x.Pricelist.Currency == evalContext.Currency.ToString());
+            }
+
+            if (evalContext.CertainDate != null)
+            {
+                query = query.Where(x => (x.StartDate == null || evalContext.CertainDate >= x.StartDate) && (x.EndDate == null || x.EndDate >= evalContext.CertainDate));
+            }
+
+            var assignments = query.ToList();
+            var assignmentsToReturn = assignments.Where(x => x.DynamicExpression == null).ToList();
+
+            foreach (var assignment in assignments.Where(x => x.DynamicExpression != null))
+            {
+                try
+                {
+                    if (assignment.DynamicExpression.IsSatisfiedBy(evalContext) && assignmentsToReturn.All(x => x.PricelistId != assignment.PricelistId))
+                    {
+                        assignmentsToReturn.Add(assignment);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "Failed to evaluate price assignment condition.");
+                }
+            }
+
+            return assignmentsToReturn.OrderByDescending(x => x.Priority).ThenByDescending(x => x.Name).Select(x => x.Pricelist);
+        }
+
+        public virtual async Task<PricelistAssignment[]> GetAllPricelistAssignments()
```

Comment:
Is this method old? Maybe add `[Obsolete]`?

Reply:
All class declared as obsolete